model/godleyIcon.cc

```diff
@@ -279,8 +279,13 @@ namespace minsky
             string::size_type start=table.cell(r,c).find_first_not_of(" ");
             if (start!=string::npos)
               {
-                FlowCoef fc(table.cell(r,c).substr(start));                      
+                FlowCoef fc(table.cell(r,c).substr(start));                                      
                 v.init=fc.str();              
+                // set initial value of stock var to init value of flow that is defined by a parameter or a constant. for ticket 1137
+                if (auto initVar=minsky().definingVar(VariableValue::valueId(group.lock(),table.cell(r,c).substr(start))))
+                  if (initVar->inputWired() && initVar->type()==VariableType::flow)
+                    if (auto lhsVar=initVar->ports[1]->wires()[0]->from()->item().variableCast())
+                       v.init=lhsVar->vValue()->init;              
```

Comment:
Code is now good, but I still don't understand why it is necessary. Also, initialisations of the form 2foo will not be handled by this code.

Reply:
I fixed the 2foo case, thanks.

I guess I've given too much attention to this ticket which actually covers very special cases. Maybe we should have changed the status of the ticket to "won't fix"?

Reply:
![2foo](https://user-images.githubusercontent.com/46541934/91273234-44d59e00-e77d-11ea-9859-614164fd53f6.png)

Reply:
I think you've missed the point of 2foo. 2foo means twice foo, so if foo's init string is 2p, then the resultant init string should be 4p.
Init strings can be parsed by the flowcoef class, so you'd want something like:
{
FlowCoef fc1(lhsVar->vValue()->init)
fc1.coef*=fc.coef;
v.init=fc1.str();
}

Reply:
Ah, I see. I thought it was just another way people could name their vars. It makes much more sense now. Thanks :)
![2foo_correct](https://user-images.githubusercontent.com/46541934/91309862-6badc780-e7b1-11ea-9b67-66cc68ab79ed.png)